test/irmin-pack/test_tree.ml

```diff
@@ -323,6 +345,25 @@ let test_small_inode () =
   let ops = [ Add ([ "00" ], ""); Del [ "01" ] ] in
   test_proofs ctxt ops
 
+let test_length_proof () =
+  let bindings = bindings fewer_steps in
+  let size = List.length fewer_steps in
+  let* ctxt = init_tree bindings in
+  let ops =
+    [
+      Length ([], size);
+      Add ([ "new" ], "");
```

Comment:
Can you also add an element that's already on the map? That's one of the tricky cases.

Reply:
Yes, done!

Reply:
"01" is in the node but not in the update map right? 

Maybe add `"new"` another time so we can test the opposite?